baybe/surrogates/utils.py

```diff
@@ -80,76 +83,94 @@ def _fit_new(self, train_x: Tensor, train_y: Tensor) -> None:
     return cls
 
 
-# FIXME[typing]: Typing should be reactivated once the `joint_posterior` attribute
-#   has been refactored/removed
-@typing.no_type_check
-def batchify(
-    posterior: Callable[[Surrogate, Tensor], tuple[Tensor, Tensor]],
-) -> Callable[[Surrogate, Tensor], tuple[Tensor, Tensor]]:
-    """Wrap ``Surrogate`` posterior functions to enable proper batching.
-
-    More precisely, this wraps model that are incompatible with t- and q-batching such
-    that they become able to process batched inputs.
-
-    Args:
-        posterior: The original ``posterior`` function.
-
-    Returns:
-        The wrapped posterior function.
-    """
+def batchify_mean_var_prediction(
+    posterior: Callable[[_TSurrogate, Tensor], tuple[Tensor, Tensor]],
+) -> Callable[[_TSurrogate, Tensor], tuple[Tensor, Tensor]]:
+    """Wrap a posterior method to make it evaluate t-batches as an augmented q-batch."""
```

Comment:
Please give a more descriptive description here. Specifically given the type hints of the arguments, it is not clear to me at all what is happening here.

Reply:
Ok, after reading through the code in more detail I get this function now. I'd still suggest to change this slighlt ysince we currently only allow a single `t`-batch dimension which is just being flattened.

Reply:
Can you comment on what you mean? Perhaps to be clear: just because we only have a single t-batch dimension doesn't mean that we only have a single t-batch. Perhaps that was the confusion?

Reply:
No, that is clear, I would have liked the docstring to reflect that we can have at most a single t-batch dimension. However, I think it will be nearly impossible to have a more detailed docstring here without blowing this up completely, so for the sake of speed, I'll just resolve here.